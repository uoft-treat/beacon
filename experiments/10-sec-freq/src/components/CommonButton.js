import React, {Component} from 'react';
import './CommonButton.scss';

export class CommonButton extends Component {
    render() {
        const {text, onClick} = this.props;
        return (
            <button className={"btn"} onClick={onClick}>
                {text}
            </button>
        )
    }
}