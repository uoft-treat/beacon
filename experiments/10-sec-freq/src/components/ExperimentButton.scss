.experiment-button {
  background-color: white;
  width: 300px;
  height: 300px;
  border-radius: 400px;
  text-align: center;
  vertical-align: center;
  line-height: 300px;
  font-size: 40px;
  display: inline-block;
  box-shadow: rgba(0,0,0,0.1) 10px 10px 40px;
  margin-top: 20px;
  margin-bottom: 20px;
  position: relative;
  top: 0;
  transition: all ease 0.3s;
  &.pressed {
    transition: all ease 0.3s;
    box-shadow: none;
    top: 5px;
  }
  &.disabled {
    box-shadow: none;
    opacity: 0.7;
    width: 100px;
    height: 100px;
    font-size: 18px;
    line-height: 100px;
    &:active {
      top: 0;
    }
  }
}