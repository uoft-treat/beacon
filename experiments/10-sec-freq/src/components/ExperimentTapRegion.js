import React from 'react';

import './ExperimentTapRegion.scss';

export const ExperimentTapRegion = ({onClick, children, disabled}) => {

    return (

        <div
            className={"experiment-tap-region" + (disabled ? " disabled" : "")}
            onClick={onClick}
        >
            {children}
        </div>

    );

};