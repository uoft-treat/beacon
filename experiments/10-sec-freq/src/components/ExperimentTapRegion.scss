.experiment-tap-region {
  background-image: url("/button.png");
  background-size: contain;
  max-width: calc(100% - 60px);
  width: 300px;
  height: 300px;
  display: inline-block;
  user-select: none;
  font-size: 30px;
  padding: 30px;
  transition: all ease 0.5s;
  &.disabled {
    width: 50px;
    height: 50px;
    font-size: 10px;
    opacity: 0.3;
    padding: 5px;
    transition: all ease 0.5s;
  }
}