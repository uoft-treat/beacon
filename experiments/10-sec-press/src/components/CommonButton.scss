.btn {
  border-radius: 4px;
  background-color: #ffffff;
  padding: 15px 25px 15px 25px;
  color: #3e260c;
  box-shadow: rgba(0,0,0,0.1) 5px 5px 10px;
  border: none;
  transition: all ease 0.3s;
  position: relative;
  left: 0;
  top: 0;
  outline: none;
  font-size: 20px;
  &:hover {
    transition: all ease 0.3s;
    box-shadow: rgba(0,0,0,0.1) 2px 2px 5px;
    left: 2px;
    top: 2px;
  }
  &:active {
    transition: all ease 0.3s;
    box-shadow: none;
    left: 5px;
    top: 5px;
  }
}