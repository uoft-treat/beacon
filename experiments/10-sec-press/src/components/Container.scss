.container {
  width: calc(100% - 40px);
  max-width: 500px;
  padding: 20px;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
}