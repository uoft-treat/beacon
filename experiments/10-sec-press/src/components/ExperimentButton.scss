.experiment-button {
  background-color: transparent;
  width: 300px;
  height: 300px;
  border-radius: 400px;
  text-align: center;
  vertical-align: center;
  line-height: 300px;
  font-size: 40px;
  display: inline-block;
  margin-top: 20px;
  margin-bottom: 20px;
  position: relative;
  top: 0;
  transition: all ease 0.3s;
  background-image: url("../images/button.png");
  background-size: contain;
  &.pressed {
    transition: all ease 0.3s;
    box-shadow: none;
    top: 5px;
  }
  &.disabled {
    box-shadow: none;
    opacity: 0.7;
    width: 100px;
    height: 100px;
    font-size: 18px;
    line-height: 100px;
    &:active {
      top: 0;
    }
  }
}