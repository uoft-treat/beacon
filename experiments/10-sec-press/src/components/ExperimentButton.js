import React from 'react';
import './ExperimentButton.scss';

export const ExperimentButton = ({onMouseDown, onMouseUp, disabled, pressed}) => {
    return (
        <div
            className={`experiment-button ${disabled && "disabled"} ${pressed && "pressed"}`}
            onMouseDown={!disabled && onMouseDown}
            onMouseUp={!disabled && onMouseUp}
            onTouchStart={!disabled && onMouseDown}
            onTouchEnd={!disabled && onMouseUp}
        />
    )
};
