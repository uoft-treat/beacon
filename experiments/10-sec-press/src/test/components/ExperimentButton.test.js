import React from 'react';
import renderer from 'react-test-renderer';
import {ExperimentButton} from "../../components/ExperimentButton";

test('Container should render according to snapshot', () => {
    const component = renderer.create(
        <ExperimentButton/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
