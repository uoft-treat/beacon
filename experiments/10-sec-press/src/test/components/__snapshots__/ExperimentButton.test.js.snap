// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render according to snapshot 1`] = `
<div
  className="experiment-button undefined undefined"
>
  Press me
</div>
`;
